package/kedro_viz/database.py

```diff
@@ -10,9 +12,27 @@ def make_db_session_factory(session_store_location: str) -> sessionmaker:
     """SQLAlchemy connection to a SQLite DB"""
     database_url = f"sqlite:///{session_store_location}"
     engine = create_engine(database_url, connect_args={"check_same_thread": False})
-    session_class = sessionmaker(engine)
     # TODO: making db session factory shouldn't depend on models.
     # So want to move the table creation elsewhere ideally.
     # But this means returning engine as well as session class.
+
+    # Check if we are running in an Azure ML environment.
+    is_azure_ml = any(
```

Comment:
Looks good :) can we move the azure logic in another function apply_azure_ml_wal_mode ?

Reply:
That makes sense!